Source/Fluids/WarpXFluidContainer.H

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2023 Grant Johnson, Remi Lehe
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_WarpXFluidContainer_H_
+#define WARPX_WarpXFluidContainer_H_
+
+#include "Evolve/WarpXDtType.H"
+#include "Initialization/PlasmaInjector.H"
+#include "MultiFluidContainer.H"
+
+#include<AMReX_MultiFab.H>
+#include<AMReX_Vector.H>
+
+#include <string>
+
+
+/**
+ * WarpXFluidContainer is the base polymorphic class from which all concrete
+ * fluid container classes derive.
+ *
+ * WarpXFluidContainer contains the main functions for initialization,
+ * interaction with the grid (field gather and current deposition), fluid
+ * source and push.
+ */
+class WarpXFluidContainer
+{
+public:
+    friend MultiFluidContainer;
+
+    WarpXFluidContainer (int nlevs_max, int ispecies, const std::string& name, const amrex::Geometry& geom);
+    ~WarpXFluidContainer() {}
+
+    void AllocateLevelMFs (int lev, const amrex::BoxArray& ba, const amrex::DistributionMapping& dm);
+
+    void InitData (int lev, amrex::Box init_box, amrex::Real cur_time);
+
+    void ReadParameters ();
+    /**
+     * Evolve updates a single timestep (dt) of the cold relativistic fluid eqautions
+     */
+    void Evolve (int lev,
+        const amrex::MultiFab& Ex, const amrex::MultiFab& Ey, const amrex::MultiFab& Ez,
+        const amrex::MultiFab& Bx, const amrex::MultiFab& By, const amrex::MultiFab& Bz,
+        amrex::MultiFab* rho, amrex::MultiFab& jx, amrex::MultiFab& jy, amrex::MultiFab& jz,
+        amrex::Real cur_time, bool skip_deposition=false);
+
+    /**
+     * AdvectivePush_Muscl takes a single timestep (dt) of the cold relativistic fluid equations
+     * using a Muscl-Handcock scheme
+     */
+    /**
+     * \brief Advective term, cold-rel. fluids
+     *
+     * \param[in] lev refinement level
+     */
+    void AdvectivePush_Muscl (int lev);
+
+
+    /**
+    * Apply (non-periodic) BC on the fluids (needed for spatial derivative),
+    * and communicate N, NU at boundaries
+    */
+    /**
+     * \brief Apply non-periodic BC to fluids and communicate boundaries
+     *
+     * \param[in] lev refinement level
+     */
+    void ApplyBcFluidsAndComms (int lev);
+
+    /**
+     * centrifugal_source adds contributions due to curvature acceleration for a
```

Comment:
Could we add `RZ` or `cylindrical` in the name of this function, to make it clear that this only applies in RZ. Also: The docstring could also have some words that indicate that the "centrifugal force" corresponds to RZ-specific terms.

Reply:
I changed the name to "centrifugal_source_rz", and updated the doc-string to mention that its specific to RZ.